src/libs/ui/src/templates/ProductListPageTemplate/index.tsx

```diff
@@ -0,0 +1,218 @@
+  const [selectedSortOption, setSelectedSortOption] = useState<string>('Alphabetical A - Z');
+  const [isBestSeller, setIsBestSellerState] = useState<boolean>(false); // Add state for isBestSeller
+  const [enableBestSeller,setEnableBestSeller]=useState<boolean>(false);
+  // Update URL params when a category is selected
+  const handleCategorySelect = (category: string | null) => {
+    const newCategory = category ?? 'all products';
+
+    // Check if the selected category is the same as the current one
+    if (newCategory === selectedProductCategory) {
+    
+      console.log(selectedProductCategory);
+    }
+
+    if (newCategory === 'best seller') {
+      setIsBestSellerState(true);
+      setFilters([]);
+      navigate('products?category=best-seller');
+      setSelectedProductCategory(newCategory); // Set the selected category
+      return;
+    }
+
+    // If "All Products" is selected, display all category badges
+    if (newCategory === 'All Products' || newCategory === 'View All') {
+      setIsBestSellerState(false); // Ensure we are not in best-seller mode
+      const allCategories = [
+        'antioxidants', 
+        'broad spectrum SPF', 
+        'Cleansers & toners', 
+        'solution sets', 
+        'eye, neck, lip', 
+        'masks', 
+        'moisturizers', 
+        'retinols', 
+        'serums'
+      ];
+      setFilters(allCategories);
+
+      // Update the URL to reflect "All Products"
+      const searchParams = new URLSearchParams(location.search);
+      searchParams.delete('category');
+      navigate({ search: searchParams.toString() });
+      setSelectedProductCategory(newCategory); // Set the selected category
+      return;
+    }
+
+    // For other specific categories
+  
+    const searchParams = new URLSearchParams(location.search);
+  
+    // Only update the category in the URL if it's different from the currently selected one
+    if (newCategory !== 'all products') {
+      searchParams.set('category', newCategory);
+    } 
+
+
+    navigate({ search: searchParams.toString() });
+  
+  };
+
+  
+
+
+  const loadMoreProducts = () => setProductsToShow(prev => prev + 9);
+
+  const breadcrumbs = useMemo(() => [
+    { label: 'Home', href: '/' },
+    { label: 'All Products', href: '/products' },
+    ...(selectedProductCategory && selectedProductCategory !== 'All Products' ? [{ label: selectedProductCategory, href: '#' }] : []),
+  ], [selectedProductCategory]);
+
+  const handleSortChange = (option: string) => {
+    setSelectedSortOption(option);
+
+    // Update the URL with the selected sort option
+    const searchParams = new URLSearchParams(location.search);
+    searchParams.set('sort', option);
+    navigate({ search: searchParams.toString() });
+  };
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any 
+  const sortProducts = (products: any[]) => {
+    switch (selectedSortOption) {
+    case 'Alphabetical A - Z':
```

Comment:
use enum or constants for swith case strings

Reply:
created constants for this